module/zfs/dsl_crypt.c

```diff
@@ -866,7 +866,7 @@ spa_keystore_unload_wkey_impl(spa_t *spa, uint64_t ddobj)
 	found_wkey = avl_find(&spa->spa_keystore.sk_wkeys,
 	    &search_wkey, NULL);
 	if (!found_wkey) {
-		ret = SET_ERROR(ENOENT);
+		ret = SET_ERROR(EACCES);
```

Comment:
This allow to differentiate the case where we tried to unload a key on a non-existing dataset (ENOENT) from the situation where a dataset has no key loaded: this is consistent with the "change" case where trying to `zfs_ioc_change_key()` from a dataset with a missing (already unloaded?) key results in EACCES:

https://github.com/zfsonlinux/zfs/blob/a76f3d0437e5e974f0f748f8735af3539443b388/module/zfs/dsl_crypt.c#L1086-L1094

@tcaputi what is your opinion on this change?

Reply:
LGTM. This was probably just an oversight on my part. Technically it is an ABI change, so we might want to check with @behlendorf in case he is aware of any consumers of this code.

Reply:
Yes looks good.  Let's go ahead and make this change now before encryption appears in a tagged release.